### PR TITLE
Add labels for `traefik.frontend.entryPoints` & `PassTLSCert` to Kubernetes

### PR DESCRIPTION
**Pull request by ryarnyah**
### Description

Add labels:
- `traefik.frontend.entryPoints`
- `traefik.frontend.passTLSCert`

### CODE REVIEW

**Comment by emilevauge** [comment]
Ping @dtomcej or @timoreimann, WDYT ?

**Comment by timoreimann** [comment]
Ah wait, this doesn't seem to be about a nee annotation, so maybe no docs needed.

Would it be worthwhile to extend the guide somehow though? @dtomcej, WDYT?

File: provider/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -165,25 +166,16 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
-					// No op.
-				case passHostHeaderAnnotation == "false":
-					PassHostHeader = false
-				case passHostHeaderAnnotation == "true":
-					PassHostHeader = true
-				default:
-					log.Warnf("Unknown value '%s' for %s, falling back to %s", passHostHeaderAnnotation, types.LabelFrontendPassHostHeader, PassHostHeader)
-				}
+				PassHostHeader := getRulePassHostHeader(i, p)
+				PassTLSCert := getRulePassTLSCert(i, p)
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: The two variables probably don't need to be exported. Can we decapitalize them?

File: provider/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -165,25 +166,16 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
 				if realm := i.Annotations[annotationKubernetesAuthRealm]; realm != "" && realm != traefikDefaultRealm {
 					log.Errorf("Value for annotation %q on ingress %s/%s invalid: no realm customization supported", annotationKubernetesAuthRealm, i.ObjectMeta.Namespace, i.ObjectMeta.Name)
 					delete(templateObjects.Backends, r.Host+pa.Path)
 					continue
 				}
 
+				EntryPoints := getEntrypoints(i)
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: Unexport this one too?

File: provider/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -199,10 +191,12 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 					templateObjects.Frontends[r.Host+pa.Path] = &types.Frontend{
 						Backend:              r.Host + pa.Path,
 						PassHostHeader:       PassHostHeader,
+						PassTLSCert:          PassTLSCert,
 						Routes:               make(map[string]types.Route),
 						Priority:             priority,
 						BasicAuth:            basicAuthCreds,
 						WhitelistSourceRange: whitelistSourceRange,
+						EntryPoints:          EntryPoints,
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: Are the entry points needed for the pass TLS functionality? If not, then I'd prefer to handle them in a separate PR.

File: provider/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -315,6 +309,48 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 	return &templateObjects, nil
 }
 
+func getEntrypoints(i *v1beta1.Ingress) []string {
+	entrypointsAnnotation, ok := i.Annotations[types.LabelFrontendEntryPoints]
+	if ok {
+		return strings.Split(entrypointsAnnotation, ",")
+	}
+	return nil
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: If the `nil` slice is handled equally to the empty slice case by Traefik, then we don't need the extra `ok` check it seems?

File: provider/kubernetes/kubernetes.go [hunk 5]
```diff
@@ -315,6 +309,48 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
+}
+
+func getRulePassHostHeader(i *v1beta1.Ingress, p *Provider) bool {
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: It's not really a _rule_ it seems to me. Can we drop that part from the function name?

File: provider/kubernetes/kubernetes.go [hunk 6]
```diff
@@ -315,6 +309,48 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
+		passHostHeader = false
+	case passHostHeaderAnnotation == "true":
+		passHostHeader = true
+	default:
+		log.Warnf("Unknown value '%s' for %s, falling back to %s", passHostHeaderAnnotation, types.LabelFrontendPassHostHeader, passHostHeader)
+	}
+	return passHostHeader
+}
+
+func getRulePassTLSCert(i *v1beta1.Ingress, p *Provider) bool {
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: Drop `Rule` from the name here too?

File: provider/kubernetes/kubernetes.go [hunk 7]
```diff
@@ -315,6 +309,48 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
+	case !ok:
+		// No op.
+	case passTLSCertAnnotation == "false":
+		passTLSCert = false
+	case passTLSCertAnnotation == "true":
+		passTLSCert = true
+	default:
+		log.Warnf("Unknown value '%s' for %s, falling back to %s", passTLSCertAnnotation, types.LabelFrontendPassTLSCert, passTLSCert)
+	}
+	return passTLSCert
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: Is the parsing logic identical to what we have in the function just above? If so, can we extract a little helper and use it from both implementation?

**Comment by ldez** [comment]
So that your PR is integrated in the 1.5, it is imperative that it is ready by Tuesday 21 November at the latest. :rocket:

**Comment by ldez** [comment]
@ryarnyah So that your PR is integrated in the 1.5, it is imperative that it is ready by Tuesday 21 November at the latest. :rocket:

File: provider/kubernetes/kubernetes.go (hunk 1 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

File: provider/kubernetes/kubernetes.go (hunk 2 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

File: provider/kubernetes/kubernetes.go (hunk 3 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Did i really need to make two separate PR?

File: provider/kubernetes/kubernetes.go (hunk 4 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: `strings.Split` return a non empty slice when his first arg is empty, so if i didn't check extra `ok` , i will need to check if first of slice is empty. It seem more maintainable that way, no?

File: provider/kubernetes/kubernetes.go (hunk 5 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

File: provider/kubernetes/kubernetes.go (hunk 6 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

File: provider/kubernetes/kubernetes.go (hunk 7 above)
Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

**Comment by ryarnyah** [comment]
I also added some docs on these.

File: provider/kubernetes/kubernetes.go (hunk 4 above)
Comment (timoreimann) [reply provider/kubernetes/kubernetes.go]: Good point, I missed that property of `strings.Split`. Let's keep things like they are right now then.

File: provider/kubernetes/kubernetes.go [hunk 8]
```diff
@@ -318,6 +312,39 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
...
+func getEntrypoints(i *v1beta1.Ingress) []string {
+	entrypointsAnnotation, ok := i.Annotations[types.LabelFrontendEntryPoints]
+	if ok {
+		return strings.Split(entrypointsAnnotation, ",")
+	}
+	return nil
+
+}
+
+func getBoolAnnoatation(meta v1.ObjectMeta, name string, defaultValue bool) bool {
```
Comment (timoreimann) [inline provider/kubernetes/kubernetes.go]: Typo: Annoatation -> Annotation.

Comment (ryarnyah) [reply provider/kubernetes/kubernetes.go]: Done.

**Pull request merged**